mlxtend/frequent_patterns/fpcommon.py

```diff
@@ -105,6 +108,11 @@ def valid_input_check(df):
     # Fast path: if all columns are boolean, there is nothing to checks
     all_bools = df.dtypes.apply(pd.api.types.is_bool_dtype).all()
     if not all_bools:
+        warnings.warn(
+            "Support for dataframe with non-bool type migth be discontinued in"
```

Comment:
A little typo: 

- migth -> might

And some little suggestions :)

I would phrase it as 

> Support for DataFrames with non-bool types result in worse computational performance and might be discontinued in the future. Please use a DataFrame with bool type.

Reply:
Thanks for the feedback! I will revise it and push again.

Reply:
@rasbt 
I think the term `Support` and `result in worse computational performance` together in one sentence is a little bit weird. What do you think?

I rephrased that as follows but I am not sure if it gets green light from you or not.
```
DataFrames with non-bool types result in worse computational performance and their support might be discontinued in the future. Please use a DataFrame with bool type"
```

Reply:
Ohh, there must have been something gone missing when I copy & pasted the two sections together 😅. I liked your rephrased version actually :)